course/resources/rqm-ecologized.md

```diff
@@ -0,0 +1,227 @@
+- As we are now incorporating some ecology classes into this schedule, I think it is important to make sure enough time is dedicated to teaching sound principles around the data analyse pipeline, as this is still the main focus of the course imo.
+- Another point of consideration is whether we should teach all the R stuff with a common data set, so that we can streamline it for teaching the principles, and then hand out student data sets after a few weeks.
+    - I am leaning towards a common data set for the first weeks, since this would make it faster and force students to go through these procedures at least twice, once with the common data set and once with their own, which is advantageous or knowledge retention. I fear classes would become choppy if every group has their own data set specific problems come up, that's better dealt with in tutorials with the group specific TAs.
+- I originally split these topics into lecture and tutorial, but I will just call it 1 and 2 instead, since both of them will hopefully be in a computer lab. The third item under each week is for suggested homework.
+- Do we want simple quizzes at the start of each lecture, for the first ~5 weeks? This could serve as an easy to grade item and make sure that the students put effort in at home. It will make it easier for the students to know where they are at the drop date, and we can then reduce the quiz frequency as the course moves into the project based phase.
+
+### Week 1
+1. Introduction to the course, intro to the ecology part, motivation behind more reproducible workflows, talk briefly about spreadsheet and get straight into how to use R.
+    - Since Martin mentioned we should expect no programming experience from these students, I want to introduce programming at the first lecture already, at least conceptually.
+    - We can do this based on a [jupyter notebook I wrote previously](https://github.com/joelostblom/resources/blob/master/from-spreadsheets-to-pandas/from-spreadsheets-to-pandas.ipynb) with analogies between spreadsheet software and programming languages. It's in Python, but if you think this is a good idea, we can adopt it for R. If not, we can do something else.
+2. Basic R programming, vectors, data frames, loops, conditions, etc.
+3. Before 2nd class: Everyone have Rstudio up and running. Maybe we can also link them to some R intro course and ask them to complete 1 or 2 lectures there per week, which will serve as the underlying material for the quiz? This one is good imo http://tryr.codeschool.com/. Watch Christie's talk https://www.youtube.com/watch?v=TthxI9mlvGE?
+
+### Week 2
+1. Basic R-programming, repetition and continued, scripting.
+2. dplyr intro, finalize R-prep by teaching outstanding items required to use R for the following ecology classes.
+3. Again, link online lectures?
+
+### Week 3
+1. Ecology background knowledge for lab 1
+2. Ecology lab 1 (or similar)
+3. Lab report? Aka send in their code.
+
+### Week 4
```

Comment:
I was unclear here. I am thinking that the first class will be mostly theory and that the second class (the lab), will be largely coding the theory the learned in the class the previous week (like martin's labs).

I agree that it can be a good idea to separate them, but we also need to think about how they will be taught. If the same person teaches both these weeks, they need to be following each other according to our current teaching plan (which we could modify of course).

And yes, extra ecology lectures might be among the more important things to get into the currently empty weeks.